smac_planner/include/smac_planner/node.hpp

```diff
@@ -27,6 +27,7 @@ namespace smac_planner
  * @class smac_planner::Node
  * @brief Node implementation for graph
  */
+template<typename T>
```

Comment:
I am a little confused what this template is. Why is the `Node` templated? I think what we're trying to do is to template the A* algorithm to use multiple `NodeT`'s, rather than having multiple data types inside of a single `Node`. Please clarify your thinking, maybe you have another valid interpretation of how to do this I didn't consider.

Reply:
I clearly misunderstood your point. When you mentioned "Template the node class" I immediately thought about actually making the class be a template based on the information it was storing. My idea was to have a generic type of Node that could hold either 2D coordinates or 2D Coordinates + orientation, although I'm currently not sure how a generic `GetNeighbors` method would work in this situation. 

After reading your comment, I now get that what you meant was to have the a_star class be a template based on the node type. With this approach we would then have two classes: node.hpp and hybrid_node (or node3d).hpp, to support both regular A* and hybrid A*. This makes sense, it may be a bit overkill to have a template node class as opposed to just making two separate classes that are compatible with the needs of a_star<NodeType>.

Reply:
yes, exactly :-) So we can utilize the exact same algorithm on a new data structure. We could also do abstract nodes, you are correct.